drivers/storage/portworx/deployment.go

```diff
@@ -361,9 +361,43 @@ func (p *portworx) GetStoragePodSpec(
 		}
 	}
 
+	p.pruneVolumes(&podSpec)
+
 	return podSpec, nil
 }
 
+func (p *portworx) pruneVolumes(spec *v1.PodSpec) {
+	seen := make(map[string]bool)
+	for ci, container := range spec.Containers {
+		seenDest := make(map[string]bool)
+		vm := container.VolumeMounts
+		for i := len(vm) - 1; i >= 0; i-- {
+			if seenDest[vm[i].MountPath] {
+				logrus.Warnf("Removed mount %s:%s for container %s due to non-unique destination",
+					vm[i].Name, vm[i].MountPath, container.Name)
+				copy(vm[i:], vm[i+1:])
+				spec.Containers[ci].VolumeMounts = vm[:len(vm)-1]
```

Comment:
this does not look correct, if there are 5 duplicate entries, this will still use len-1. suggest use a different array to save the elements instead of changing array inside of iteration.

Reply:
This should be working also for cases when all 5 entries are duplicates:
* we are traversing the array right-to-left
* if we need to remove an element from the array, we'll copy the elements and "trim" the last element from the array
* the code "activates" only when the duplicate entry has been identified -- so we won't go "out of array bounds" with `[:len(vm)-1]`

Granted, the way this is done is not very efficient  (element's `copy` takes time), and we could achieve similar effect by building a brand new array as a replacement.  However, please consider:
* the code does not need to be very efficient, as the occurrence of the volume-mount removal should be _exceedingly rare_
* building a replacement array (`spec.Containers[#].VolumeMounts`) every time would be inefficient, since the elements are full struct's (note, they are not pointers to struct's)


So, sadly -- golang does not have a more elegant way how to remove elements from the arrays, and I wrote my reasons why I think doing "in place" element removals is more appropriate here.
Is this (still) going to be an issue?

Reply:
Did not want to nit-pick here, but now that we are discussing this, even I feel creating a separate array is better for two reasons - easy to understand and more time efficient than copying array every time. I understand it is not space efficient but there are not a lot of volumes either.

Reply:
@zoxpx I got that, for example if there are 5 duplicates, the array should have one in the end, but this assignment will save 4 because it uses len(vm) which is 5, did you want to trim the last element after copying the array one element ahead?

spec.Containers[ci].VolumeMounts = vm[:len(vm)-1]

Reply:
@haibinxie -- sorry, still not sure where the problem is.  Note that with the trimming at the line you pointed out, we are changing the array _slice_ without modifying the array.  This would happen 4 times if there are 5 duplicates, but the array "re-slicing" should be very cheap in golang.
If your concern that the backing array will still be 5 elements long, where the "useful" array slice is only 1 element?


@piyush-nimbalkar -- please check the 3953cb8 for updated code:
* this version uses a new array instead of "in-place" elements removal
* note that now we're probably copying more data, also we have to revert the "replacement" mounts' array since the elements were added in reverted order  (e.g. "last guy wins")

Reply:
i see it is fixed in new commit.